### DISCUSSION TITLE
Storage Class migration for Brokers and Zookeepers

### DISCUSSION THREAD

**Discussion by MandMaric** [strimzi/strimzi-kafka-operator#10104] [Q&A]
We would like to migrate the storage class for all our brokers and zookeepers. Currently, we are using Strimzi 0.36.0.

The requirement we have is to perform the migration without any data loss on all our environments including PROD.

Is there a best practice to do that? I saw that there is a new feature with KafkaNodePools (https://strimzi.io/blog/2023/08/28/kafka-node-pools-storage-and-scheduling/) but it is not production ready yet.

**Comment by scholzj** [comment]
For Kafka, you have several options ... you can use KafkaNodePools or JBOD storage. NodePools are production-ready in recent versions (not entirely sure about 0.36, you might need to upgrade). 

For ZooKeeper, there is no easy way how to do it without downtime. There are various _hacky_ ways how to do it. There are several discussions (for sure on Slack, but I think also here) going through the various options.
